From 4a49c45559bdd144a775da40446a20f16fc19ba0 Mon Sep 17 00:00:00 2001
From: Peter Xu <peterx@redhat.com>
Date: Thu, 10 Mar 2016 14:37:32 +0800
Subject: [PATCH 058/134] trace-cmd,bash: Add auto-complete for bash

This is the most simplified version. Only pop up sub-cmds and nothing
else.

Signed-off-by: Peter Xu <peterx@redhat.com>
Signed-off-by: Steven Rostedt <rostedt@goodmis.org>
---
 trace-cmd.bash | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 trace-cmd.bash

diff --git a/trace-cmd.bash b/trace-cmd.bash
new file mode 100644
index 000000000000..e7e4673ddc5d
--- /dev/null
+++ b/trace-cmd.bash
@@ -0,0 +1,13 @@
+_trace_cmd_complete()
+{
+    local sub_cmd_list
+    sub_cmd_list=$(trace-cmd --help 2>/dev/null | \
+                          grep " - " | sed 's/^ *//; s/ -.*//')
+    COMPREPLY=()
+    prev="${COMP_WORDS[COMP_CWORD-1]}"
+    cur="${COMP_WORDS[COMP_CWORD]}"
+    if [[ "$prev" == "trace-cmd" ]]; then
+        COMPREPLY=( $(compgen -W "${sub_cmd_list}" -- ${cur} ))
+    fi
+}
+complete -F _trace_cmd_complete trace-cmd
-- 
2.5.5

